tests/testthat/test_move_pieces.R

```diff
@@ -552,4 +552,75 @@ testthat::test_that(
         result <- apply_type_1_move(gudir, r, b)
         testthat::expect_true(igraph::isomorphic(expected, result)) 
     }
-)
\ No newline at end of file
+)
+
+
+######################################
+###### validate_type_3_move() ########
+######################################
+
+testthat::test_that(
+    "Test that validate_type_3_move() returns FALSE when there is a non-empty intersection",
+    {
+        gdir <- igraph::graph_from_edgelist(
+            matrix(c(
+                c(1,2),
+                c(4,3),
+                c(2,3)
+            ), ncol = 2, byrow = TRUE), directed = TRUE)
+        
+        b_u <- igraph::graph_from_edgelist(
+            matrix(c(
+                c(1,3),
+                c(2,3)
+            ), ncol = 2, byrow = TRUE), directed = FALSE)
+        
+        result <- validate_type_3_move(gdir, b_u)
+        testthat::expect_false(result)
+    }
+)
+
+######################################
+###### generate_type_3_move() ########
+######################################
+
+testthat::test_that(
+    "Test that generate_type_3_move() returns correct output on valid example",
+    {
+        #skip("Incomplete test")
+
+        gdir <- igraph::graph_from_edgelist(
```

Comment:
Change example to a valid graph

Reply:
Since generate_type_3_move() sequentially applies type 1 and type 2 moves, I applied the valid Type 1 move to the undirected part before feeding it into generate_type_2_move() to avoid rejecting valid moves.